parquet-column/src/main/java/org/apache/parquet/column/values/deltastrings/DeltaByteArrayWriter.java

```diff
@@ -82,11 +82,11 @@ public String memUsageString(String prefix) {
   @Override
   public void writeBytes(Binary v) {
     int i = 0;
-    byte[] vb = v.getBytes();
+    byte[] vb = v.getBytesUnsafe();
     int length = previous.length < vb.length ? previous.length : vb.length;
     for(i = 0; (i < length) && (previous[i] == vb[i]); i++);
     prefixLengthWriter.writeInteger(i);
-    suffixWriter.writeBytes(Binary.fromByteArray(vb, i, vb.length - i));
+    suffixWriter.writeBytes(Binary.fromUnmodifiedByteArray(vb, i, vb.length - i));
```

Comment:
This looks odd. The contract for unsafe access to the bytes backing another Binary is to use them immediately and forget about them so it doesn't matter if they change. But in this case you create a constant Binary that signals the bytes won't change. So this is taking something that might change and passing on something that won't.

What we need here is the ability to get a slice of the original v that inherits whether it is reused or constant:

```java
Binary slice(int start, int length);
```